rero_ils/modules/users/validators.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2023 RERO
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Password validator."""
+
+from wtforms import ValidationError
+
+from rero_ils.modules.utils import PasswordValidatorException, \
+    password_validator
+
+
+class PasswordValidator:
+    """Password Validator."""
+
+    def __init__(self, length=8, special_char=False):
+        """Password validator class initializer."""
+        self.length = length
+        self.special_char = special_char
+
+    def __call__(self, form, field):
+        """Call.
+
+        :param form: the current form.
+        :param field: the password field.
+        :raise ValidationError: If the password isn't valid.
+        """
+        field_length = field.data and len(field.data) or 0
```

Comment:
Seems strange... `field.data` should return boolean. So your test `field.data and len(field.data)` will compare a boolean and an integer ; this integer will be casted to boolean. So `field_length` should return `True|False` (impossible to return 0).
Next line, you compare `field_length < self.length`, so the boolean value is now casted to int (0|1).

A solution should be to use the ugly python ternary operator : `field_length = len(field.data) if field.data else 0`